src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportProcessingJob.cs

```diff
@@ -83,7 +83,8 @@ async Task<ExportJobOutcome> UpdateExportJob(ExportJobRecord exportJobRecord, We
                 record = exportJobRecord;
                 progress.Report(JsonConvert.SerializeObject(exportJobRecord));
 
-                if (record.Status == OperationStatus.Running)
+                if (record.Status == OperationStatus.Running
```

Comment:
I think this statement may be needed for system level export as well (without filters). I tested this for perf and system level export execution went into this block to queue a new job w/ the continuation token / throw a `JobSegmentException` to set the current job in memory to complete.

This logic is opaque - does this change also work for system level exports without filters?

Reply:
That shouldn't have been happening. In system level export each job should only be handling one page of data, there shouldn't be a continuation token.
Ug, this may be another partitioning thing.

Reply:
Could be mistaken but this is my mental model from the perf work.

In the new Queue Client implementation,`CosmosExportOrchestratorJob.cs` kicks off a single Export Job for a system level export that may have tens millions of resources. I didn't see any code in the **Cosmos** side to split this up. SQL side does this great.

The Cosmos side was working with the new queue client implementation by creating a single export record and then iterating (mostly single worker doing something at once) over the result until there is no continuation token. The `JobSegmentException` was crucial to this - I remember my logs filling up like crazy.

Reply:
Ah, you are correct. I was thinking of the SQL pattern, I forgot we had setup Cosmos to just use one job.